### PR TITLE
executor: add correctness tests about direct reading with aggregations

### PR DESCRIPTION
**Pull request by zhuo-zhi**
### What problem does this PR solve?

Issue Number: related to #24150 

Problem Summary:

executor: add correctness tests about direct reading with aggregations

### What is changed and how it works?

What's Changed:

Add some correctness tests about partition table direct reading functors with different aggregation functors.

How it Works:

N/A

### Check List 

Tests 

- Unit test

Side effects

- N/A

### Release note 

- executor: add correctness tests about direct reading with aggregations

### CODE REVIEW

File: executor/partition_table_test.go [hunk 1]
```diff
@@ -305,6 +305,150 @@ func (s *partitionTableSuite) TestGlobalStatsAndSQLBinding(c *C) {
...
+	// test range partition
+	for i := 0; i < 2000; i++ {
+		// select /*+ stream_agg() */ a from t where a > ? group by a;
+		// select /*+ hash_agg() */ a from t where a > ? group by a;
+		// select /*+ stream_agg() */ a from t where a in(?, ?, ?) group by a;
+		// select /*+ hash_agg() */ a from t where a in (?, ?, ?) group by a;
+		x := rand.Intn(1099)
+
+		queryPartition1 := fmt.Sprintf("select /*+ stream_agg() */ a from trange where a > %v group by a;", x)
+		queryRegular1 := fmt.Sprintf("select /*+ stream_agg() */ a from tregular1 where a > %v group by a;", x)
```
Comment (qw4990) [inline executor/partition_table_test.go:364]: Please add some aggregation functions for the like `sum(b), max(b)...`~

**Comment by qw4990** [comment]
Almost LGTM, should we remove the `WIP` label? @zhuo-zhi

File: executor/partition_table_test.go (hunk 1 above)
Comment (zhuo-zhi) [reply executor/partition_table_test.go]: Now it's updated.

**Comment by tisonkun** [comment]
This pull request seems part of #24150, not close it. Please verify the case and update description on demand.

**Review by qw4990 — APPROVED** [review]

**Review by rebelice — APPROVED** [review]
/lgtm

**Pull request merged**